packages/base-driver/lib/basedriver/capabilities.js

```diff
@@ -221,8 +225,41 @@ function processCapabilities (caps, constraints = {}, shouldValidateCaps = true)
   return matchedCaps;
 }
 
+/**
+ * Return a copy of a capabilities object which has taken everything within the 'options'
+ * capability and promoted it to the top level. Note that this function is assumed to be run after
+ * all vendor prefixes have already been stripped from the top level. So we are dealing with e.g.
+ * 'options' and not 'appium:options' at this point. Any prefixes _inside_ the 'options' capability
+ * will themselves be stripped. This is designed as an internal function, not one to operate on
+ * user-constructed capabilities.
+ *
+ * @param {object} originalCaps - the capabilities to analyze and promote from 'options'
+ * @return {object} - the capabilities wit 'options' promoted if necessary
+ */
+function promoteAppiumOptions (originalCaps) {
+  const appiumOptions = originalCaps[APPIUM_OPTS_CAP];
+  if (!appiumOptions) {
+    return originalCaps;
+  }
+
+  let caps = _.cloneDeep(originalCaps);
+  if (!_.isPlainObject(appiumOptions)) {
+    throw new errors.SessionNotCreatedError(`The ${APPIUM_OPTS_CAP} capability must be an object`);
+  }
+
+  // first get rid of any prefixes inside appium:options
+  stripAppiumPrefixes(appiumOptions);
+
+  // now just apply them to the main caps object
+  caps = {...caps, ...appiumOptions};
```

Comment:
should we allow to override caps silently if they already present in the root object?

Reply:
I would say so, yes. appium:options will be more specific.

Reply:
i guess we could warn in a log statement